main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1724,33 +1725,47 @@ object Weeder2 {
       }
     }
 
+    private def visitRunExpr(tree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
+      expect(tree, TreeKind.Expr.Run)
+      val maybeWith = pickAll(TreeKind.Expr.TryWithBodyFragment, tree)
+      flatMapN(
```

Comment:
Does this mirror the structure of visitTryExpr?

Reply:
Yes